sdk/lib/opentelemetry/sdk/resources/auto_detector.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/sdk/resources/detectors/google_cloud_platform'
+require 'opentelemetry/sdk/resources/detectors/telemetry'
+
+module OpenTelemetry
+  module SDK
+    module Resources
+      # AutoDetector contains detect class method for running all detectors
+      module AutoDetector
+        extend self
+
+        DETECTORS = [
+          OpenTelemetry::SDK::Resources::Detectors::GoogleCloudPlatform,
+          OpenTelemetry::SDK::Resources::Detectors::Telemetry
+        ].freeze
+
+        def detect
+          resources = DETECTORS.map(&:detect)
+          resources.reduce(OpenTelemetry::SDK::Resources::Resource.create) do |empty_resource, detected_resource|
```

Comment:
Any chance we could rename `empty_resource`? It is likely non-empty at some point and is really the accumulated resource. I don't have a specific name in mind. Two names I often use, which are completely generic are `memo`, or `acc`.